libs/pdi-scanner/examples/scan_forever.rs

```diff
@@ -75,98 +72,103 @@ async fn main() -> color_eyre::Result<()> {
     // Sometimes, after closing the previous scanner connection, a new connection will
     // time out during these first commands. Until we get to the bottom of why that's
     // happening, we just retry once, which seems to resolve it.
-    if client
-        .send_initial_commands_after_connect(Duration::from_millis(500))
-        .is_err()
+    if timeout(
+        Duration::from_millis(500),
+        client.send_initial_commands_after_connect(),
+    )
+    .await
+    .is_err()
     {
-        client.send_initial_commands_after_connect(Duration::from_secs(3))?;
+        timeout(
+            Duration::from_secs(3),
+            client.send_initial_commands_after_connect(),
+        )
+        .await??;
     }
-    let image_calibration_tables = client.get_image_calibration_tables(Duration::from_secs(3))?;
-
-    client.send_enable_scan_commands(
-        config.bitonal_threshold,
-        DoubleFeedDetectionMode::RejectDoubleFeeds,
-        11.0,
-    )?;
+    let image_calibration_tables = timeout(
+        Duration::from_secs(3),
+        client.get_image_calibration_tables(),
+    )
+    .await??;
+
+    client
+        .send_enable_scan_commands(
+            config.bitonal_threshold,
+            DoubleFeedDetectionMode::RejectDoubleFeeds,
+            11.0,
+        )
+        .await?;
     println!("waiting for sheet…");
 
-    let running = Arc::new(AtomicBool::new(true));
-
-    ctrlc::set_handler({
-        let running = running.clone();
-        move || {
-            eprintln!("received SIGINT");
-            running.store(false, Ordering::SeqCst);
-        }
-    })?;
     let mut start: Option<Instant> = None;
 
     loop {
-        if !running.load(Ordering::SeqCst) {
-            eprintln!("exiting");
-            break;
-        }
+        tokio::select! {
```

Comment:
This makes it so we don’t have to busy loop like we did before. Now we simply await multiple `Future`s and `tokio::select!` picks whichever one is ready first. Also means we can drop the `ctrlc` dependency.